history-server.md

```diff
@@ -0,0 +1,76 @@
+# Spark History Server
+
+### Prerequisites
+
+Required software:
+* K8s cluster
+* [KUDO CLI Plugin](https://kudo.dev/docs/#install-kudo-cli) 0.7.3 or higher
+
+### Installing Spark Operator with History Server Enabled
+
+```
+kubectl kudo install ./kudo-operator \
+    -n <NAMESPACE> \
+    -p operatorImageName=mesosphere/kudo-spark-operator \
+    -p operatorVersion=latest \
+    -p enableHistoryServer=true \
+    -p historyServerFsLogDirectory="s3a://<BUCKET_NAME>/<FOLDER>" \
+    -p historyServerOpts="-Dspark.hadoop.fs.s3a.access.key=<AWS_ACCESS_KEY_ID> -Dspark.hadoop.fs.s3a.secret.key=<AWS_SECRET_ACCESS_KEY> -Dspark.hadoop.fs.s3a.impl=org.apache.hadoop.fs.s3a.S3AFileSystem"
+```
+
+This will deploy a Pod and Service for the `Spark History Server` with the `Spark Event Log` directory configured via the `historyServerFsLogDirectory` parameter. This is an S3 backed storage for event logs. Spark Operator also supports Persistent Volume Claim (PVC) based storage. There is a parameter `historyServerPVCName` to pass the name of the PVC. Make sure that provided PVC should have `ReadWriteMany` [access mode](https://kubernetes.io/docs/concepts/storage/persistent-volumes/#access-modes) supported.
+
+### Creating Spark Application
+
+Make sure `specs/spark-application.yaml` has these configurations present under `spec.sparkConf`.
+
+```
+"spark.eventLog.enabled": "true"
+"spark.eventLog.dir": "s3a://<BUCKET_NAME>/<FOLDER>"
+"spark.hadoop.fs.s3a.access.key": "<AWS_ACCESS_KEY_ID>"
+"spark.hadoop.fs.s3a.secret.key": "<AWS_SECRET_ACCESS_KEY>"
+```
+
+If PVC is passed while installing the Spark Operator, make sure these two fields are also added with following values:
+
+```
+# Add this under SparkApplicationSpec
+Volumes:
+- name: pvc-storage
+  persistentVolumeClaim:
+    claimName: <HISTORY_SERVER_PVC_NAME>
+
+# Add this under SparkPodSpec
+volumeMounts:
+- mountPath: <HISTORY_SERVER_FS_LOG_DIRECTORY>
+  name: pvc-storage
+```
+
+Make sure that CRD and RBAC for SparkApplication is already created. Now we can run the application as follows:
+
+```
+kubectl apply -n <NAMESPACE> -f specs/spark-application.yaml
+```
+
+### Accessing Spark History Server UI
+
+You can run this command to expose the UI in your local machine.
+
+`kubectl port-forward <HISTORY_SERVER_POD_NAME> 18080:18080`
+
+Verify local access using this: `curl -L localhost:18080`
+
+
+### List of available parameters for Spark History Server
```

Comment:
👍 a great example of user-focused documentation

Reply:
Thanks